src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java

```diff
@@ -49,14 +53,39 @@
     //Bases below this quality will not be included in picking the best read from a set of duplicates.
     private static final int MIN_BASE_QUAL = 15;
 
+    // Used to set an attribute on the GATKRead marking this read as an optical duplicate.
+    private static final String OPTICAL_DUPLICATE_ATTRIBUTE_NAME = "OD";
+
     @Argument(doc="output BAM file", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
     protected String outputFile;
 
     @Argument(doc = "uri for the input bam, either a local file path or a gs:// bucket path",
-            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
-            optional = false)
+              shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+              optional = false)
     protected String bam;
 
+    @Argument(doc = "File to write duplication metrics to.", optional=true,
+              shortName = "M", fullName = "METRICS_FILE")
+    protected File metricsFile;
+
+    @Argument(doc = "Regular expression that can be used to parse read names in the incoming SAM file. Read names are " +
+            "parsed to extract three variables: tile/region, x coordinate and y coordinate. These values are used " +
+            "to estimate the rate of optical duplication in order to give a more accurate estimated library size. " +
+            "Set this option to null to disable optical duplicate detection. " +
+            "The regular expression should contain three capture groups for the three variables, in order. " +
+            "It must match the entire read name. " +
+            "Note that if the default regex is specified, a regex match is not actually done, but instead the read name " +
+            " is split on colon character. " +
+            "For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values. " +
+            "For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.",
+            optional = true)
+    public String READ_NAME_REGEX = OpticalDuplicateFinder.DEFAULT_READ_NAME_REGEX;
```

Comment:
Could you add a test case with null here?  We're probably going to be changing our handling of null in input at some point and it would be good to know that we'll catch this instance if we do.